src/binaryen-c.h

```diff
@@ -103,6 +103,7 @@ typedef void* BinaryenFunctionTypeRef;
 // Add a new function type. This is thread-safe.
 // Note: name can be NULL, in which case we auto-generate a name
 BinaryenFunctionTypeRef BinaryenAddFunctionType(BinaryenModuleRef module, const char* name, BinaryenType result, BinaryenType* paramTypes, BinaryenIndex numParams);
+BinaryenFunctionTypeRef BinaryenGetFunctionType(BinaryenModuleRef module, BinaryenType result, BinaryenType* paramTypes, BinaryenIndex numParams);
```

Comment:
Sure, but the problem would remain if the optimizer wasn't part of the plan, i.e. because the user actually intends to keep unused functions so these can be used in the next build step (i.e. after linking something else). Currently, there's only either _use the optimizer with default options_ or _don't use the optimizer at all_ when using binaryen.js.

I personally felt that adding GetFunctionType (or GetStructuralFunctionType, or GetFunctionTypeBySignature), using pretty much the same parameters as AddFunctionType, would make a good API addition especially because it doesn't require a shift in how to obtain an existing function type.